lib/src/rules/avoid_init_to_null.dart

```diff
@@ -89,7 +100,9 @@ class _Visitor extends SimpleAstVisitor<void> {
     if (!node.isConst &&
         !node.isFinal &&
         DartTypeUtilities.isNullLiteral(node.initializer)) {
-      rule.reportLint(node);
+      if (isNullable(node.declaredElement.type)) {
```

Comment:
The enclosing `if` combines multiple checks with `&&`.
So, probably this additional check could be added there.

Same for the similar check for parameters above.